[jeffrey-jian] iP

jeffrey-jian opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [X]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jeffrey-jian committed:
Rename JED, greet and exit

jeffrey-jian committed:
Echo user input

jeffrey-jian committed:
Add list function

jeffrey-jian committed:
Use Tasks

jeffrey-jian committed:
Add mark and unmark features

jeffrey-jian committed:
Add ToDos

jeffrey-jian committed:
Add Deadlines

jeffrey-jian committed:
bugfix string trim

jeffrey-jian committed:
simple todo bye test

jeffrey-jian committed:
More comprehensive test

jeffrey-jian committed:
Refactor code, use switch case

jeffrey-jian committed:
Add DukeException for error handling

jeffrey-jian committed:
Add exceptionhandling to test

jeffrey-jian committed:
Bug fix, working app

jeffrey-jian committed:
Use ArrayList for tasks

jeffrey-jian committed:
Add delete

jeffrey-jian committed:
Add test for delete

jeffrey-jian committed:
fix indent for switch-case

jeffrey-jian committed:
Read and write to duke.txt file

jeffrey-jian committed:
Merge branch 'branch-Level-7'

jeffrey-jian committed:
Use datetime for Deadline

jeffrey-jian committed:
Use datetime for event

jeffrey-jian committed:
Merge branch 'branch-Level-8'

jeffrey-jian committed:
use Ui class

jeffrey-jian committed:
use Storage class

jeffrey-jian committed:
use TaskList

jeffrey-jian committed:
use Parser and Command classes

jeffrey-jian committed:
Merge branch 'more-oop'

jeffrey-jian committed:
A-Packages

jeffrey-jian committed:
Merge branch 'packages'

jeffrey-jian committed:
Merge branch 'add-gradle-support'

jeffrey-jian committed:
Fix build.grade mainClass name

jeffrey-jian committed:
Add JUnit ToDoTest

jeffrey-jian committed:
Add JUnit Parser test

jeffrey-jian committed:
Merge branch 'junit'

jeffrey-jian committed:
Follow coding standard, remove unnecessary imports

jeffrey-jian committed:
JavaDoc for Duke, Ui

jeffrey-jian committed:
JavaDoc for task

jeffrey-jian committed:
JavaDoc for Storage, Parser, DukeException

jeffrey-jian committed:
JavaDoc for commands

jeffrey-jian committed:
Level-9

jeffrey-jian committed:
Merge branch 'codingStandard'

jeffrey-jian committed:
Merge branch 'branch-Level-9'

jeffrey-jian committed:
JavaDocs for Ui done

jeffrey-jian committed:
Resolve merge conflicts, add find javadocs

jeffrey-jian committed:
Add config/checkstyle/ files

jeffrey-jian committed:
Adhere to checkstyle

jeffrey-jian committed:
Merge branch 'add-checkstyle'

jeffrey-jian committed:
update build.gradle javafx dependencies

jeffrey-jian committed:
Basic app with textField and sendButton

jeffrey-jian committed:
Add dummy Duke responses and avatar pics

jeffrey-jian committed:
WIP connect javafx to ui, working exceptions

jeffrey-jian committed:
Ui commands now return text

jeffrey-jian committed:
command.execute now returns duke response

jeffrey-jian committed:
Fix MainWindow controller

jeffrey-jian committed:
Merge branch 'command-return-text' into javafx

jeffrey-jian committed:
Merge branch 'javafx', working GUI

jean-cq commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,149 @@
+            } catch (DateTimeParseException e) {
+                throw new DukeException(DukeException.WRONG_DATETIME_MESSAGE);
+            }
+        case "event":
+            if (commandArr.length < 2) {
+                throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+            }
+            description = commandArr[1];
+            String eventName = "";
+            String eventFrom = "";
+            String eventTo = "";
+            String[] eventDescription = description.split("/");
+            for (String str : eventDescription) {
+                if (str.startsWith("from")) {
+                    eventFrom = str.split(" ", 2)[1].trim();
+                } else if (str.startsWith("to")) {
+                    eventTo = str.split(" ", 2)[1].trim();
+                } else {
+                    eventName = str.trim();
+                }
+            }
+            if (eventName.equals("")) {
+                throw new DukeException("OOPS!! Please include the name of the event.");
+            }
+            if (eventFrom.equals("")) {
+                throw new DukeException("OOPS!! Please include when the event is from.");
+            }
+            if (eventTo.equals("")) {
+                throw new DukeException("OOPS!! Please include when the event is till.");
+            }
+            try {
+                LocalDateTime fromParsed = LocalDateTime.parse(eventFrom, DATETIME_INPUT_FORMAT);
+                LocalDateTime toParsed = LocalDateTime.parse(eventTo, DATETIME_INPUT_FORMAT);
+                return new EventCommand(eventName, fromParsed, toParsed);
+            } catch (DateTimeParseException e) {
+                throw new DukeException(DukeException.WRONG_DATETIME_MESSAGE);
+            }
+        default:
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```
You treat the switch case very carefully in terms of coding standard.

jean-cq reviewed (commented):
LGTM.
Just a little reminder to keep the commit summary in format.

kiatkat commented on src/data/duke.txt:
Should this file be version controlled?

kiatkat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,85 @@
+        }
+    }
+
+    public String getResponse(String input) {
```
Is this method missing a Javadoc?

kiatkat commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
Perhaps consider using a more descriptive name here? Similarly with Event.

kiatkat commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,147 @@
+     *
+     * @param items The ArrayList of searched items to be printed.
+     */
+    public String listSearch(ArrayList<?> items) {
```
Why does the `items` parameter here need to be of type `ArrayList<?>`? Specifically confused about the need for the wildcard generic in the ArrayList.

kiatkat reviewed (commented):
Looks good overall! Just a few minor points.

jeffrey-jian committed:
Add duke.txt data file to gitignore

jeffrey-jian committed:
Add assert to task constructors

Assertions are added to verify assumptions about the
Internal Invariants, Control-Flow Invariants, Preconditions,
Postconditions, and Class Invariants.

In this case, we are asserting that the arguments passed into
the constructors of task commands should not be null.

jeffrey-jian committed:
Refactor to improve code quality

Refactored some variables to be declared as final, as
they will not be changed.

jeffrey-jian committed:
Merge pull request #1 from jeffrey-jian/branch-A-Assertions

Add assert to task constructors

jeffrey-jian committed:
Merge pull request #2 from jeffrey-jian/branch-A-CodeQuality

Refactor to improve code quality

jeffrey-jian committed:
Add yml file for CI

jeffrey-jian committed:
Merge branch 'branch-A-CI'

jeffrey-jian committed:
set macos as only platform in gradle.yml

jeffrey-jian committed:
set macos as platform in gradle.yml

jeffrey-jian committed:
Add Alias support

jeffrey-jian committed:
Merge branch 'branch-C-FriendlierSyntax'

jeffrey-jian committed:
fix checkstyle

jeffrey-jian committed:
Add greeting, removed unnecessary ui methods

jeffrey-jian committed:
Add color to dialogBox

jeffrey-jian committed:
Apply styling to dialog

jeffrey-jian committed:
Merge branch 'A-BetterGui'

jeffrey-jian committed:
Move task commands to task package

jeffrey-jian committed:
AddAlias done

jeffrey-jian committed:
ListAlias done

jeffrey-jian committed:
DeleteAlias done

jeffrey-jian committed:
set title as JEDuke

jeffrey-jian committed:
Complete User Guide, add Ui.png

jeffrey-jian committed:
Add auto create data file
